### ISSUE TITLE
planned improvements

### ISSUE THREAD

**Issue by martindurant** [fsspec/kerchunk#106]
- combine to support direct writing to remote
- support multiple filesystems in ReferenceFileSystem (will be great for multi-resolution, where we can put downsampled images somewhere other than originals)
- more hooks for custom functions to manage zarr objects and/or post-process output JSON

**Comment by joshmoore** [comment]
> * I may have mentioned this idea

I definitely remember the discussion, I think from the Oct. 20th community call. One of the examples was chunking for 365 days versus 366 in leap years. I don't have a feeling for the implications though.

**Comment by martindurant** [comment]
> I don't have a feeling for the implications though.

I'm pretty sure very light. In the case that the chunks is a list of lists (dask style), we use a copy of dask's indexing to figure out which slices of which chunks are required. If the chunks is a simple list (all existing data sets), we use the current code.

**Comment by rsignell-usgs** [comment]
@martindurant , this may already be covered in your list, but I'd like to be able to combine variables that are in different files, to avoid notebooks like this: https://nbviewer.org/gist/rsignell-usgs/1f8f74c3e4c7023d5b919444e42fd23a

**Comment by martindurant** [comment]
@rsignell-usgs , yes combine2 does this, but I'm not convinced I started in the right direction there. I am glad, though, that the JSONs are simple enough that post-processing can workaround shortcomings in the current code, however painful.
